### PR TITLE
Add / update some redirects for old URLs and unify samples

### PR DESCRIPTION
**Pull request by thaJeztah**
Cleaning up branches on my machine, and found this one

See individual commits for details

### CODE REVIEW

**Comment by thaJeztah** [comment]
@usha-mandya ptal

File: samples/django.md [hunk 1]
```diff
@@ -270,9 +272,9 @@ In this section, you set up the database connection for Django.
...
-- [User guide](index.md)
-- [Installing Compose](install.md)
-- [Getting Started](gettingstarted.md)
-- [Command line reference](reference/index.md)
-- [Compose file reference](compose-file/index.md)
-- [Sample apps with Compose](samples-for-compose.md)
+- [Docker Compose overview](../compose/index.md)
+- [Install Docker Compose](../compose/install.md)
+- [Getting Started with Docker Compose](../compose/gettingstarted.md)
+- [Docker Compose Command line reference](../compose/reference/index.md)
```
Comment (usha-mandya) [inline samples/django.md:278]: The Compose Command line reference page appears as a floating page as it is not linked from the toc.

**Comment by usha-mandya** [comment]
@thaJezta Thanks for the PR. Left a comment about adding the Compose CLI reference page to the repo. Looks good otherwise.

**Comment by thaJeztah** [comment]
> The Compose Command line reference page appears as a floating page as it is not linked from the toc.

@usha-mandya good catch; I removed the old "index.md", and moved the "overview.md" to take its place, now the compose reference index is at https://docs.docker.com/compose/reference/, instead of https://docs.docker.com/compose/reference/overview/

**Comment by usha-mandya** [comment]
That's a good idea.

**Review by usha-mandya — APPROVED** [review]
LGTM

**Pull request merged**